spacy/training/initialize.py

```diff
@@ -23,6 +24,9 @@
 def init_nlp(config: Config, *, use_gpu: int = -1) -> "Language":
     raw_config = config
     config = raw_config.interpolate()
+    # Validate config before accessing values
+    _validate_config_block(config, "initialize", ConfigSchemaInit)
+    _validate_config_block(config, "training", ConfigSchemaTraining)
```

Comment:
I don't think we can really do this here because it means we're creating objects multiple times and the config is resolved twice. That's really unideal at runtime.

Instead, we should just wrap the calls to `registry.resolve` (maybe using a context manager,  similar to `cli._util.show_validation_error`). If we do this here and in `Language.initialize`, the result should be the same – the error may just be raised slightly later (which is an okay tradeo-ff IMO).